src/System.Net.Primitives/tests/FunctionalTests/NetworkCredentialTest.netstandard1.7.cs

```diff
@@ -0,0 +1,107 @@
+            yield return new object[] { "", AsSecureString("") };
+            yield return new object[] { null, AsSecureString(null) };
+        }
+
+        [Fact]
+        public static void Ctor_SecureString_Test()
+        {
+            string expectedUser = "UserName";
+
+            using (SecureString expectedSecurePassword = AsSecureString("password"))
+            {
+                NetworkCredential nc = new NetworkCredential(expectedUser, expectedSecurePassword);
+                Assert.Equal(expectedUser, nc.UserName);
+                Assert.True(expectedSecurePassword.CompareSecureString(nc.SecurePassword));
+            }
+        }
+
+        [Fact]
+        public static void Ctor_SecureStringDomain_Test()
+        {
+            string expectedUser = "UserName";
+            string expectedDomain = "thisDomain";
+
+            using (SecureString expectedSecurePassword = AsSecureString("password"))
+            {
+                NetworkCredential nc = new NetworkCredential(expectedUser, expectedSecurePassword, expectedDomain);
+                Assert.Equal(expectedUser, nc.UserName);
+                Assert.Equal(expectedDomain, nc.Domain);
+                Assert.True(expectedSecurePassword.CompareSecureString(nc.SecurePassword));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(SecurePassword_TestData))]
+        public static void SecurePasswordSetGet_Test(string password, SecureString expectedPassword)
+        {
+            NetworkCredential nc = new NetworkCredential();
+            using (SecureString securePassword = AsSecureString(password))
+            {
+                nc.SecurePassword = securePassword;
+                Assert.True(expectedPassword.CompareSecureString(nc.SecurePassword));
+            }
+        }
+
+        private static bool CompareSecureString(this SecureString s1, SecureString s2)
+        {
+            if(s1 == null || s2 == null)
+                return false;
+
+            IntPtr ptr1 = IntPtr.Zero;
+            IntPtr ptr2 = IntPtr.Zero;
+            string str1 = string.Empty;
+            string str2 = string.Empty;
+            try
+            {
+                ptr1 = Marshal.SecureStringToGlobalAllocUnicode(s1);
+                ptr2 = Marshal.SecureStringToGlobalAllocUnicode(s2);
+                str1 = Marshal.PtrToStringUni(ptr1);
+                str2 = Marshal.PtrToStringUni(ptr2);
+            }
+            finally
+            {
+                if(ptr1 != IntPtr.Zero)
+                {
+                    Marshal.ZeroFreeGlobalAllocUnicode(ptr1);
+                }
+                if(ptr2 != IntPtr.Zero)
+                {
+                    Marshal.ZeroFreeGlobalAllocUnicode(ptr2);
+                }
+            }
+            return str1 == str2;
+        }
+
+        private static SecureString AsSecureString(string str)
+        {
+            SecureString secureString = new SecureString();
+
+            if(str == null)
```

Comment:
I did it like this because of the style rule 1 on corefx guidelines... Style: (Rule 1): "A single line statement block can go without braces but the block must be properly indented on its own line and it must not be nested in other statement blocks that use braces."

@CIPop should I add it to the same line?